compiler/hash-ast/src/ast.rs

```diff
@@ -876,9 +876,11 @@ define_tree! {
     #[node]
     pub struct RangePat {
```

Comment:
Maybe add a comment "Invariant : must have at least one of lo, hi"

Reply:
But you don't actually need this

Reply:
What's the point of omitting both?

Reply:
Well it is the same as just specifying a wildcard, but I don’t see much point in disallowing it.